android/source/VideoLocker/src/org/edx/mobile/view/MyCoursesListActivity.java

```diff
@@ -323,4 +337,51 @@ public void onFail(Exception ex) {
             logger.error(ex);
         }
     };
+
+    /**
+     * Checks progress of all the videos that are being downloaded.
+     * If progress of any of the downloads is 100%, then marks the video as DOWNLOADED.
+     */
+    private void doRepairData() {
+        Thread maintenanceThread = new Thread(new Runnable() {
```

Comment:
Is this a one time migration or is it something that needs to happen regularly? This seems like it might have threading issues if the user hits back and then shows the screen again. Will there be a problem if this operation is running multiple times overlapping? Also, it's a bunch of database logic. It probably doesn't want to live inside the activity itself.

Does getApplicationContext() do the right thing if the activity is torn down on one thread before the repair starts?